pkg/cloud/talos/provisioners/packet/packet.go

```diff
@@ -0,0 +1,227 @@
+}
+
+type Network struct {
+	OS OS
+}
+
+type OS struct {
+	Devices []map[string]interface{}
+}
+
+//NewPacket returns an instance of the Packet provisioner
+func NewPacket() (*Packet, error) {
+	c, err := packngo.NewClient()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return &Packet{client: c}, nil
+}
+
+// Create creates an instance in Packet.
+func (packet *Packet) Create(cluster *clusterv1.Cluster, machine *clusterv1.Machine, clientset *kubernetes.Clientset) error {
+
+	clusterSpec, err := utils.ClusterProviderFromSpec(cluster.Spec.ProviderSpec)
+	if err != nil {
+		return err
+	}
+
+	machineSpec, err := utils.MachineProviderFromSpec(machine.Spec.ProviderSpec)
+	if err != nil {
+		return err
+	}
+
+	packetConfig := &ClusterInfo{}
+	yaml.Unmarshal([]byte(machineSpec.Platform.Config), packetConfig)
+
+	// Here we pull down the userdata config map and add the install section to the end if it's defined in the machine
+	udConfigMap, err := utils.FetchConfigMap(cluster, machine, clientset)
+	if err != nil {
+		return err
+	}
+	udStruct := &Userdata{}
+	yaml.Unmarshal([]byte(udConfigMap.Data["userdata"]), udStruct)
+
+	if packetConfig.Instances.Install != nil {
+		udStruct.Install = packetConfig.Instances.Install
+	}
+
+	//Add network tweaks for elastic IPs to userdata
+	floatingIP := ""
+	isMaster := strings.Contains(machine.ObjectMeta.Name, "master")
+	if isMaster {
+		nameSlice := strings.Split(machine.ObjectMeta.Name, "-")
+		indexString := nameSlice[len(nameSlice)-1]
+		index, err := strconv.Atoi(indexString)
+		if err != nil {
+			return err
+		}
+		floatingIP = clusterSpec.Masters.IPs[index]
+
+		//Haxx on haxx b/c dhcp value needs to be a bool so the whole networking.os.devices block needs to be a map of interfaces
+		udStruct.Networking.OS.Devices = append(udStruct.Networking.OS.Devices, map[string]interface{}{"interface": "lo", "cidr": floatingIP + "/32"})
+		udStruct.Networking.OS.Devices = append(udStruct.Networking.OS.Devices, map[string]interface{}{"interface": "eth0", "dhcp": true})
+	}
+
+	udBytes, err := yaml.Marshal(udStruct)
+	if err != nil {
+		return err
+	}
+	//TODO(rsmitty): Shebang no longer needed once talos alpha 28 is cut.
+	ud := "#!talos\n" + string(udBytes)
+
+	devCreateReq := &packngo.DeviceCreateRequest{
+		Hostname:      machine.ObjectMeta.Name,
+		Plan:          packetConfig.Instances.Plan,
+		Facility:      []string{packetConfig.Instances.Facility},
+		OS:            "custom_ipxe",
+		BillingCycle:  "hourly",
+		ProjectID:     packetConfig.ProjectID,
+		UserData:      ud,
+		IPXEScriptURL: packetConfig.Instances.PXEUrl,
+	}
+
+	dev, _, err := packet.client.Devices.Create(devCreateReq)
+	if err != nil {
+		return err
+	}
+
+	//Wait for masters to be active, attach floating ip
+	if isMaster {
+		err = packet.waitForStatus(machine, "active")
+		if err != nil {
+			return err
+		}
+		ipReq := &packngo.AddressStruct{Address: floatingIP + "/32"}
+		_, _, err = packet.client.DeviceIPs.Assign(dev.ID, ipReq)
```

Comment:
Do we need to do a detach event or will the above handle that automatically?

Reply:
Need to explore this. Will create backlog ticket to make sure the floating IP just gets move automatically. The way the docs are written, I feel like it should.